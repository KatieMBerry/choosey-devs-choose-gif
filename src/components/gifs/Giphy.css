.Giphy {
    text-decoration: none;
    color: black;
    cursor: pointer;

    figure {
        width: 15rem;
        text-align: center;
        text-transform: uppercase;

    img {
        width: 100%;
    }
    }
}
