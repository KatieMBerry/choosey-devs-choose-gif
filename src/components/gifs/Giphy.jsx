import React from 'react';
import PropTypes from 'prop-types';
import styles from './Giphy.css';

const Giphy = ({ title, url, images }) => (
    <a className={styles.Giphy} href={url}>
        <figure>
            <img src={images} alt={title} />
            <figcaption>{title}</figcaption>
        </figure>
    </a>
);

Giphy.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    images: PropTypes.string.isRequired
};

export default Giphy;
