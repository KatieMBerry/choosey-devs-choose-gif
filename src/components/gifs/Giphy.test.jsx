import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Giphy from './Giphy';

describe('Giphy component', () => {
    afterEach(() => cleanup());
    it('renders a single GIF', () => {
        const { asFragment } = render(<Giphy
            title='winnie the pooh dogs GIF'
            url='https://giphy.com/gifs/dogs-honey-winnie-the-pooh-3ohc13lGnOGpMHn7zO'
            images='https://media1.giphy.com/media/3ohc13lGnOGpMHn7zO/giphy-downsized-medium.gif?cid=697f7defd7fyrfz51nirurnkx3ydqt7h7y6l42ymo9ie2ksx&rid=giphy-downsized-medium.gif&ct=g'
        />);
        expect(asFragment()).toMatchSnapshot();
    })
})
