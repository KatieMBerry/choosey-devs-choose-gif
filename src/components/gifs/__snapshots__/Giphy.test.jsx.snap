// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Giphy component renders a single GIF 1`] = `
<DocumentFragment>
  <a
    class="undefined"
    href="https://giphy.com/gifs/dogs-honey-winnie-the-pooh-3ohc13lGnOGpMHn7zO"
  >
    <figure>
      <img
        alt="winnie the pooh dogs GIF"
        src="https://media1.giphy.com/media/3ohc13lGnOGpMHn7zO/giphy-downsized-medium.gif?cid=697f7defd7fyrfz51nirurnkx3ydqt7h7y6l42ymo9ie2ksx&rid=giphy-downsized-medium.gif&ct=g"
      />
      <figcaption>
        winnie the pooh dogs GIF
      </figcaption>
    </figure>
  </a>
</DocumentFragment>
`;
